presto-main/src/test/java/com/facebook/presto/operator/project/TestPageProcessor.java

```diff
@@ -355,6 +373,38 @@ public void testYieldProjection()
         assertFalse(output.hasNext());
     }
 
+    @Test
+    public void testExpressionProfiler()
+    {
+        CallExpression add10Expression = call(
+                internalOperator(ADD, BIGINT.getTypeSignature(), ImmutableList.of(BIGINT.getTypeSignature(), BIGINT.getTypeSignature())),
+                BIGINT,
+                field(0, BIGINT),
+                constant(10L, BIGINT));
+
+        PageFunctionCompiler functionCompiler = new PageFunctionCompiler(createTestMetadataManager(), 0);
+        Supplier<PageProjection> projectionSupplier = functionCompiler.compileProjection(add10Expression, Optional.empty());
+        PageProjection projection = projectionSupplier.get();
+        Page page = new Page(createLongSequenceBlock(1, 10));
+        ExpressionProfiler profiler = new ExpressionProfiler(10, 10_000);
```

Comment:
I think you can create better tests, if you change this class to use a `Ticker` and then you can control exactly how time moves in the test.